sql/core/src/test/scala/org/apache/spark/sql/errors/QueryCompilationErrorsSuite.scala

```diff
@@ -979,6 +979,39 @@ class QueryCompilationErrorsSuite
     )
   }
 
+  test("trailing comma in select statement") {
+
+    intercept[ParseException] {
+      sql("SELECT 1,")
+    }
+
+    withTable("t1") {
+      sql(s"CREATE TABLE t1 (c1 INT, c2 INT) USING PARQUET")
+
+      val queries = Seq(
+        "SELECT *? FROM t1",
+        "SELECT c1? FROM t1",
+        "SELECT c1? FROM t1 WHERE c1 = 1",
+        "SELECT c1? FROM t1 GROUP BY c1",
+        "SELECT *, RANK() OVER (ORDER BY c1)? FROM t1",
+        "SELECT c1? FROM t1 ORDER BY c1",
+        "WITH cte AS (SELECT c1? FROM t1) SELECT * FROM cte",
+        "WITH cte AS (SELECT c1 FROM t1) SELECT *? FROM cte",
+        "SELECT * FROM (SELECT c1? FROM t1)",
```

Comment:
I think you need soem sanity tests that FROM i still supported.
E.g. SELECT c1, from FROM VALUES(1, 2) AS T(c1, from);
DECLARE VARIABLE from = 5;
SELECT c1, from t;